test/lib/docker.test.ts

```diff
@@ -66,6 +67,55 @@ test("docker run", async (t) => {
   });
 });
 
+test("save from docker daemon", async (t) => {
+  const targetImage = "openjdk-test:latest";
+  const docker = new Docker(targetImage);
+  await subProcess.execute("docker", [
+    "load",
+    "--input",
+    path.join(
+      __dirname,
+      "../fixtures/docker-archives",
+      "docker-save/openjdk.tar",
+    ),
+  ]);
+
+  t.test("image saved to specified location", async (t) => {
+    const destination = path.join(os.tmpdir(), "image.tar");
+
+    t.tearDown(() => {
+      if (existsSync(destination)) {
+        unlinkSync(destination);
+      }
+    });
+
+    await docker.save(targetImage, destination);
+    t.true(existsSync(destination));
+  });
+
+  t.test("promise rejects when image doesn't exist", async (t) => {
+    const image = "someImage:latest";
+    await t.rejects(
+      docker.save(image, "tmp/image.tar"),
+      "server error",
+      "rejects with expected error",
+    );
```

Comment:
how about an assertion no new file is created?

Reply:
Sure!